### PR TITLE
feat: register page with api

### PR DESCRIPTION
**Pull request by femacedogomes**
Task:
https://www.notion.so/028386dd8cc04ffeaa06931d912bd0fc?v=03cd2ce2d5c848bba9d72d3f836b25cd&p=3a5e026044ff4d078ca7996eb40eae28&pm=s

### CODE REVIEW

File: src/pages/auth/register.tsx [hunk 1]
```diff
@@ -34,17 +39,55 @@ const Register = () => {
     setEmailInvalid(false);
     setEmail(e.target.value);
   };
+  const handleToast = (label: string, type?: TostifyType) => {
+    UseTostify({ label, type });
+  };
```
Comment (renanpontez) [inline src/pages/auth/register.tsx]:
```suggestion
  const handleToast = (label: string, type?: TostifyType) => UseTostify({ label, type });
```

File: src/pages/auth/register.tsx [hunk 2]
```diff
@@ -34,17 +39,55 @@ const Register = () => {
...
+      const response = await UserService.register({
+        name,
+        email,
+        password,
+      });
+      setIsLoading(false);
+      if (!response.user) {
+        handleToast('Email already taken.', 'warning');
+        return;
+      }
```
Comment (renanpontez) [inline src/pages/auth/register.tsx:55]:
```suggestion
      if (!response.user) {
        return handleToast('Email already taken.', 'warning');
      }
```

File: src/pages/auth/register.tsx [hunk 3]
```diff
@@ -64,7 +107,7 @@ const Register = () => {
               >
                 <Input
                   onChange={userNameChange}
-                  errorMessage={`userName should be 2-29 characters without number or special character!`}
+                  errorMessage={`Username should be 2-29 characters without special character!`}
```
Comment (renanpontez) [inline src/pages/auth/register.tsx]: Remember: that's not a userName. That's the User's name.

`Felipe Macedo !== felipinhomacedo77`

So we can ask like: "Oops, looks like that's not your real name."

**Review by renanpontez — APPROVED** [review]
No blocking issues. You can change the suggestion and merge. Nice

File: src/pages/auth/register.tsx (hunk 1 above)
Comment (femacedogomes) [reply src/pages/auth/register.tsx]: i removed "{}" but eslint still putting in two different lines.

**Pull request merged**